### ISSUE TITLE
Добавить память в ENPRG6695

### ISSUE THREAD

**Issue by snegin** [forth32/qtools#12]
Возможно ли добавить флеш Hynix_H9DA1GH51HBMBR в этот загрузчик?

**Comment by vvevvevve** [comment]
Найдите в загрузчике блок с описанием какой-нибудь похожей по геометрии флешки. содержащий её идентификатор. Его можно найти по строке с моделью флешки. А потом замените в нём идентификатор "той, похожей" флешки на тот, который у "Вашей, нужной".

**Comment by snegin** [comment]
А как определить идентификатор своей?Есть блок из какого-то загрузчика  в котором есть эта память,но он отличается от тех что в патченных загрузчиках.Я так понимаю идентификатор написан немного перед названием самой флеши?

**Comment by vvevvevve** [comment]
Нет. В "здешнем" ENPRG6695 он находится по смещению 0x14 от начала строки с названием. Это два или 4 байта, первый из которых - идентификатор производителя (0xad для Hynix), второй определяет объём, напряжение питания и разрядность шины данных флешки. Если интересно, посмотрите код функции get_flash_config() в qcio.c.

**Comment by vvevvevve** [comment]
Ну и, конечно, идентификаторы можно узнать в документации (даташите) на микросхему.

**Comment by snegin** [comment]
Спасибо.Изучу и попробую.

**Comment by snegin** [comment]
Добавил ид своей флешки,но загрузчик не запускается.Как и раньше.Висит на запуске загрузчика.

**Comment by vvevvevve** [comment]
Значит, дело не в идентификаторах, а в чём-то другом. 
Что за железка-то?

**Comment by snegin** [comment]
Huawei EC315.У устройств на одном проце адрес загрузки у всех должен быть один и тот же?Запускается загрузчик ENPRG6695_GOBI3000_p ,а в ответ на хеллоу тишина.

**Comment by vvevvevve** [comment]
Конечно, один и тот же. 
Этот модем предназначен для продажи в розницу или "операторский"? Если верно второе, то для какого оператора он предназначен?
Я просто вспоминаю старые хуавеевские железки вроде E153, некоторые операторские модификации которых чем-то отличались от остальных. Может быть, там SB включен, или ещё что-то в этом роде. Но ни одна прошивка, работающая на "нормальной" модификации, вообще даже не запускалась вот на таких, "странных". В то же время, глядя одновременно на платы обеих модификаций, отличий между ними заметить невозможно.

Да, а Вы его - этот модем - разбирали? Там точно MDM6600, как говорят люди?

**Comment by snegin** [comment]
Конечно,разбирал)6600.Только что разобрал такой же - один в один.Проц и память одинаковые.NPRG работает отлично.Стёр мибиб -pbl - отправил загрузчик - отлично пишется/читается.Так что аппаратное там.Относительно хуавеев встречал что NPRG не работает - решение простое - загнать в pbl.После этого всё отлично работает.

**Comment by snegin** [comment]
Конечно не в иде там дело.И адрес правильный и загрузчики пробовал.РАМ вероятно мёртвый или не контакт.Надо будет попробовать поменять память.Хотя бывало что и проц такую "красоту" делал.

**Comment by vvevvevve** [comment]
Да таким железкам уже лет 7 как место на помойке. Неужели есть какой-то резон заниматься их аппаратным ремонтом?

**Comment by snegin** [comment]
Практически спортивный)Хотя у нас ещё продаются.Цена нового около 37$.Есть места на карте где проблема с инетом - цена/качество.

**Comment by vvevvevve** [comment]
Ёлки ж палки! $37 за такую железку...
Впрочем, CDMA-оборудование никогда не было сильно дешёвым (по сравнению с аналогичными моделями без поддержки CDMA) за пределами Северной Америки. В масштабах всего мира (за исключением той же Америки) это очень редкий сейчас стандарт.
Правда, сейчас во многих современных телефонах эта поддержка идёт автоматически, "до кучи", порой даже без упоминания о ней.

**Comment by snegin** [comment]
Индия,Китай - сейчас одни из основных поставщиков по сдма.Из Китая в основном смартфоны,Индия - недорогие роутеры/модемы.Все на 6600.Из США немного и более современное.Самое последнее что было Netgear 791 на 9635.Работает довольно хорошо,аккумулятор супер,поддерживает все кроме сдма450.Смартфоны сдма можно купить в несколько раз дешевле аналогичных по характеристикам джсм-ок.Даже если смотреть по айфону 4 джсм и сдма - первый дороже где-то в 2 раза.
